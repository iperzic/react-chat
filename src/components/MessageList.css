.MessageList {
    max-width: 640px;
    margin: 0 auto;
    padding-bottom: 68px;
}
