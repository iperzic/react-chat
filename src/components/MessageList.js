import React from 'react';
import PropTypes from 'prop-types';

import './MessageList.css';

import MessageBubble from './MessageBubble';

const MessageList = ({ messages }) =>
  (
    <div className="MessageList">
      {messages.map(m =>
        (<MessageBubble
          key={m.id}
          author={m.author}
          message={m.message}
          timestamp={m.timestamp}
          isOwn={m.isOwn}
        />))
      }
    </div>
  );


MessageList.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.shape({
    author: PropTypes.string.isRequired,
    message: PropTypes.string.isRequired,
    timestamp: PropTypes.string.isRequired,
    isOwn: PropTypes.bool.isRequired,
  })).isRequired,
};

export default MessageList;
