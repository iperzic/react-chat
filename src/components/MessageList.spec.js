import React from 'react';
import { shallow } from 'enzyme';

import MessageList from './MessageList';

describe('MessageList', () => {
  let component;
  let messages;

  describe('when there are no messages', () => {
    beforeEach(() => {
      messages = [];
      component = shallow(<MessageList
        messages={messages}
      />);
    });

    it('should not display any messages', () => {
      expect(component.getElement()).toMatchSnapshot();
    });
  });

  describe('when there are messages provided', () => {
    beforeEach(() => {
      messages = [{
        id: 'abc', author: 'Bugs Bunny', message: 'Whats up doc?', timestamp: 'some timestamp', isOwn: true,
      }, {
        id: 'abcd', author: 'Foo', message: 'Bar', timestamp: 'Baz', isOwn: false,
      }];
      component = shallow(<MessageList
        messages={messages}
      />);
    });

    it('should display messages', () => {
      expect(component.getElement()).toMatchSnapshot();
    });
  });
});
