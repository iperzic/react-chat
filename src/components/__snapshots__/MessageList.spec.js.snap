// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList when there are messages provided should display messages 1`] = `
<div
  className="MessageList"
>
  <MessageBubble
    author="Bugs Bunny"
    isOwn={true}
    message="Whats up doc?"
    timestamp="some timestamp"
  />
  <MessageBubble
    author="Foo"
    isOwn={false}
    message="Bar"
    timestamp="Baz"
  />
</div>
`;

exports[`MessageList when there are no messages should not display any messages 1`] = `
<div
  className="MessageList"
>
  
</div>
`;
